### PR TITLE
CLOUDP-135435: AtlasCLI: Add ARM64 builds to Yum & apt for supported platforms

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-135435

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

This needs to be merged to confirm if it works, we need to be prepared for a possible rollback

### CODE REVIEW

File: build/ci/release.yml [hunk 1]
```diff
@@ -256,7 +256,7 @@ functions:
         aws_secret: ${aws_secret}
         local_files_include_filter:
           - src/github.com/mongodb/mongodb-atlas-cli/tasks.json
-        remote_file: ${project}/dist/${revision}_${created_at}/
+        remote_file: ${project}/dist/${revision}_${created_at}/${tool_name}/${task_type}/
```
Comment (gssbzn) [inline build/ci/release.yml:259]: we were overwriting the generated jsons, last write wins

File: tools/genevergreen/main.go [hunk 2]
```diff
@@ -60,11 +76,153 @@ func buildDependency(toolName, os, serverVersion, repo string) shrub.TaskDepende
...
+	}
+}
+
+type Platform struct {
+	extension     string
+	architectures []string
+}
+
+// if updating this list verify build/ci/repo_config.yaml matches.
+var distros = map[string]Platform{
```
Comment (fmenezes) [inline tools/genevergreen/main.go:94]: nit: move all vars together

**Review by fmenezes — APPROVED** [review]
LGTM

File: tools/genevergreen/main.go (hunk 2 above)
Comment (gssbzn) [reply tools/genevergreen/main.go:94]: going to merge and if I don't need to revert I'll do a follow up

**Pull request merged**